packages/relay-compiler/generic/core/RelayWatchmanClient.js

```diff
@@ -17,6 +17,15 @@ const watchman = require('fb-watchman');
 class RelayWatchmanClient {
   _client: any;
 
+  static createIfAvailable(): Promise<?RelayWatchmanClient> {
+    return new Promise((resolve) => {
+      const client = new RelayWatchmanClient();
+      client.on('error', () => resolve(null));
```

Comment:
Let's try out your current implementation and adjust from there if anything goes wrong. Thanks for opening the issue on watchman, hopefully someone who knows more about it than I do can help us out.